dex-it/src/test/scala/com/wavesplatform/it/sync/api/ws/WsAddressStreamTestSuite.scala

```diff
@@ -416,5 +417,27 @@ class WsAddressStreamTestSuite extends WsSuiteBase with TableDrivenPropertyCheck
         wsc.receiveAtLeastN[WsAddressState](1)
       }
     }
+
+    "DEX-817 Invalid balances after connection (leasing)" in {
+      val bobWavesBalanceBefore = dex1.api.tradableBalance(bob, wavesBtcPair)(Waves)
+
+      dex1.stopWithoutRemove()
+      val leaseTx = mkLease(bob, alice, bobWavesBalanceBefore - 0.1.waves, fee = leasingFee)
+      broadcastAndAwait(leaseTx)
+      dex1.start()
+
+      val wsc = mkDexWsConnection(dex1)
+      wsc.send(WsAddressSubscribe(bob, WsAddressSubscribe.defaultAuthType, mkJwt(bob)))
+
+      eventually {
+        val balance = wsc.receiveAtLeastN[WsAddressState](1).map(_.balances).squashed - btc - wct
+        balance should matchTo(
+          Map[Asset, WsBalances](
```

Comment:
Maybe we can automate this widespread cast somehow?

Reply:
Probably by `def mkBalanceChanges(xs: (Assets, Long)*)`. See no reason to do this, it's the only case.